### ISSUE TITLE
[New build - EXPERIMENTAL] 23w52a: Better Hovercrafts, Hotfixes

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#5394]
The hovercraft repulsor logic has been overhauled, it now feels much closer to the suspension of a car.
The ride will feel much less stiff and more natural. 
If set very soft it will be wobbly for a nice hovercraft feel.

Hovercraft changes:
 - New spring stiffnes and damping setting.
 - Some tooltip help have been added to the setting
 - Hovercrafts can now ride on the planet water too.
 - incorrect sfx origin issue has also been fixed.

Also some much needed crash fixes related to crafting.

Bugfixes:
#5350 Wrong symmetry on some rounded bricks
#5320 Crashes will crafting
Missing chunk of terrain

**Comment by TIKIRobo** [comment]
![image](https://github.com/tsunamayo/Starship-EVO/assets/56371294/b7ad6cec-bc83-480c-9a39-c18465903eaa)
unfortunately it seems like the missing chunks are still around 

https://github.com/tsunamayo/Starship-EVO/assets/56371294/3aaf1826-72d7-4844-8079-396994d45055

also seems to be some sort of second planet surface not sure what causes it, maybe fast moving close to the surface? 


https://github.com/tsunamayo/Starship-EVO/assets/56371294/8141a9fb-351c-4897-a8ae-88b990e85383

this video isn't great but it does show 2 minor issues with hovers,
1. they seem to hug the terrain better, this would be good except often times the hover shoots down, slams into the ground and then is launched back up again and flips, i believe this is the cause of the small flip at the beginning of the video a small bump that caused the hover to freak out and mess up. its quite a common issue when doing jumps over small craters etc
2. towards the end of the video i try tilting my bike to see if the anti roll function would try and push it back up, it did not

also small suggestion if you're doing hovercraft tweaks, do you think you could make the hover settle down when they are turned off? they have a kind of warm-up when they start but when turned off they just drop which makes my bikes fall over even when their kickstands are deployed  

https://github.com/tsunamayo/Starship-EVO/assets/56371294/b48fdc78-23c1-4ae7-a987-883b36ab641c
at 100% spring strength

https://github.com/tsunamayo/Starship-EVO/assets/56371294/5dc16339-d333-4e2e-8e65-041ca18470bc
at 40% strength
the bike in both examples (altho more strongly at 100%) pulls down with great force slamming into the other crater wall, i understand this may be something that is difficult to fine tune but it would be nice if the bike only closely followed the terrain if the slope was say 30 degrees or less, that way it wouldn't try and follow the steep incline down, however i leave that up to your better judgment as i have no idea how to fine tune that without more testing

**Comment by Gallt** [comment]
@TIKIRobo 

I feel hovercraft having much slower dip/drift after a ramp/drop, and maybe even an invisible "training wheels" so it doesnt tip too easy (invisible collision with the ground only 3m to the left and right of the boundaries of the vehicle) will help tremendously too. was gonna suggest it a while ago but this articulates it much better. Flipping a hover bike with the magnet/flip tool is a hassle as is: so maybe only having any invisible "training wheels" work while somone is riding on it would be best, or have flip/mag tool be much better?

**Comment by TIKIRobo** [comment]
im not sure about training wheels it would kinda defeat the purpose of bikes being inherently unstable but a better flip tool and better anti roll would be nice

**Comment by ProPeach** [comment]
Always good to see hovers get some love! I definitely agree with the direction of aiming for car suspension. Really cool to see the return of hovering on water too.

However, I can't seem to achieve a good balance of settings in this patch - The previous hovercraft behaviour seemed to absorb the small bumps on terrain better, while in this patch I can't get a ride that isn't affected by every small bump in the planet surface. You can see the car jerking very frequently.


https://github.com/tsunamayo/Starship-EVO/assets/27857711/262dbd98-e5eb-4b45-9ffc-7ec48dd0f117


And the settings for this clip - I tried lots of other settings but couldn't tone down the jerkiness much more without compromising the other aspects of the ride.
![image](https://github.com/tsunamayo/Starship-EVO/assets/27857711/ce5ee827-874f-4312-83e5-97271fb58273)
The previous hover behaviour seemed to smooth out these micro-bumps better, although it was a lot more floaty so you would end up flying more often than in this patch, so that's an improvement for sure.

Here's the bp for the car, it's unfinished so the yoke and core are under the front - [AE86.zip](https://github.com/tsunamayo/Starship-EVO/files/13772069/AE86.zip)


Also just a small note regarding the VFX, the dust cloud seems to be misplaced compared the the thrust direction an the hover pods. The dust should be further out to the side, not directly down from the pods.
![image](https://github.com/tsunamayo/Starship-EVO/assets/27857711/6090da52-9d7e-45cc-a583-83af958b3a67)

**Comment by tsunamayo** [comment]
Hi, all thanks for the feedback, I will look into it. Dont hesitate to share your blueprint. 
Now actually I already started to work on the revamp of the push tool, it is now an anti-grav tool like the magnet tool from Breath of the Wild (you can pick stuff and move stuff around).

**Comment by tsunamayo** [comment]
@TIKIRobo Are you on AMD? This fix was for a recent regression with a few terrain chunk missing or misplaced, not the one on AMD. I still need to order an AMD card for it, I just want to make sure I will be able to run a dual GPU setup.

**Comment by TIKIRobo** [comment]
@tsunamayo yep on an AMD card, glad to hear you're still working on it tho!
https://steamcommunity.com/sharedfiles/filedetails/?id=3039616295 
https://steamcommunity.com/sharedfiles/filedetails/?id=3019834895
also here are some of my hover bikes, I haven't updated them yet with the new core settings but if you need any tiny hovers for testing they do exist lol

**Comment by tsunamayo** [comment]
@TIKIRobo Hi, so first your hover are amazing ;)
So the I improved the drop-off, it is smoother now, but I still cant land your tricky one. I think it just too unstable, even with the grab tool I cant place it upside up easily. I think a set of two landing gear at the back should fix it.

**Comment by TIKIRobo** [comment]
@tsunamayo not sure what you mean by tricky one but ive found that my hovers can generally right themselves (so long as its on its side) if you hold Q/E to lean which lets the hover pods "catch" the ground and push you back up 
the drop off is a great improvement tho thank you so much!
